### PR TITLE
feat: move most of the plugins to new backend system

### LINKED ISSUE #1294: support the new backend system for http-scaffolder-plugin
support the new backend system for http-scaffolder-plugin

### PR DESCRIPTION
**Pull request by drodil**
Closes #1294

#### :heavy_check_mark: Checklist

- [ ] Added tests for new functionality and regression tests for bug fixes
- [x] Added changeset (run `yarn changeset` in the root)
- [ ] Screenshots of before and after attached (for UI changes)
- [ ] Added or updated documentation (if applicable)

### CODE REVIEW

**Comment by kissmikijr** [comment]
Do we need a version upgrade to 1.24 as well to support the backend system ? or not much changes and makes sense to do it together? It looks like a lot of code changes right now and not sure what related to what exactly :D

**Comment by drodil** [comment]
Yes we do as for example the scaffolder mock action context is not available earlier. Also some exports missing from backstage/backend-plugin-api before 1.24.

**Comment by drodil** [comment]
And yes, it's quite a big change and doesn't include even all of the plugins. So some work still required.

File: .changeset/wise-needles-provide.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+---
```
Comment (Xantier) [inline .changeset/wise-needles-provide.md:1]: Most of our plugins are past 1.x so there should be some `minor` bumps in here

**Comment by kissmikijr** [comment]
bumped the plugin's versions to the backstage version 1.25.0 so this PR shouldn't to handle that anymore

**Comment by punkle** [comment]
@drodil this is great work. I am just realizing that I may have stood on your toes with the backend integration for the rag-ai-backend yesterday. Sorry about that, I had intended to get a feel for the process and didn't realise you had actually started the work.

What can we do to help you get this merged?

**Comment by drodil** [comment]
Probably should abandon this and do the migration one plugin at a time. Otherwise it's just too much 😅

**Comment by punkle** [comment]
@drodil sounds like a good plan!

**Comment by drodil** [comment]
@punkle sounds like a good idea!

**Comment by drodil** [comment]
This PR could be used as an example/reference for the migration. But I will close this now.

**Comment by hairmare** [comment]
:point_right: https://github.com/RoadieHQ/roadie-backstage-plugins/issues/1325

**Pull request closed without merging**